### PR TITLE
Laboratory work #1, Nataliya Bushueva - 21FPL2

### PR DESCRIPTION
**Pull request by NataliaBush**
Testing the system

### CODE REVIEW

**Review by artyomtugaryov — COMMENTED** [review]
Hello, @NataliaBush! As a reminder, the deadline for the first lab is September 30th and you have less than 2 weeks to complete it. Please hurry to be on time.

**Comment by NataliaBush** [comment]
> 

the work is done

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -3,108 +3,164 @@
...
-    """
-    pass
+       Removes punctuation, casts to lowercase, splits into tokens
+       Parameters:
+       text (str): Original text
+       Returns:
+       list[str]: A sequence of lowercase tokens with no punctuation
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(text, str):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:18]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -3,108 +3,164 @@
...
-    Parameters:
-    text (str): Original text
-
-    Returns:
-    list[str]: A sequence of lowercase tokens with no punctuation
-
-    In case of corrupt input arguments, None is returned
-    """
-    pass
+       Removes punctuation, casts to lowercase, splits into tokens
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:11]: Revert the docstring, please

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -3,108 +3,164 @@
...
-    pass
+       Excludes stop words from the token sequence
+       Parameters:
+       tokens (List[str]): Original token sequence
+       stop_words (List[str]: Tokens to exclude
+       Returns:
+       List[str]: Token sequence that does not include stop words
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(tokens, list) and isinstance(stop_words, list):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:42]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py [hunk 4]
```diff
@@ -3,108 +3,164 @@
...
+        Parameters:
+        frequencies (Dict): A dictionary with tokens and
+        its corresponding frequency values
+        top (int): Number of token to extract
+        Returns:
+        List[str]: Sequence of specified length
+        consisting of tokens with the largest frequency
+        In case of corrupt input arguments, None is returned
+        """
+    if (isinstance(top, int) and top > 0 and isinstance(top, bool) is not True and isinstance(frequencies, dict)
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:86]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py [hunk 5]
```diff
@@ -3,108 +3,164 @@
...
-    """
-    pass
+       Composes a frequency dictionary from the token sequence
+       Parameters:
+       tokens (List[str]): Token sequence to count frequencies for
+       Returns:
+       Dict: {token: number of occurrences in the token sequence} dictionary
+       In case of corrupt input arguments, None is returned
+       """
+    if isinstance(tokens, list):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:63]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py [hunk 6]
```diff
@@ -3,108 +3,164 @@
...
+       Calculates Term Frequency score for each word in a token sequence
+       based on the raw frequency
+       Parameters:
+       frequencies (Dict): Raw number of occurrences for each of the tokens
+       Returns:
+       dict: A dictionary with tokens and corresponding term frequency score
+       In case of corrupt input arguments, None is returned
+       """
+    term_freq = {}
+    if isinstance(frequencies, dict):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:110]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py [hunk 7]
```diff
@@ -3,108 +3,164 @@
...
+        based on its TF and IDF scores
+        Parameters:
+        term_freq (Dict): A dictionary with tokens and its corresponding TF values
+        idf (Dict): A dictionary with tokens and its corresponding IDF values
+        Returns:
+        Dict: A dictionary with tokens and its corresponding TF-IDF values
+        In case of corrupt input arguments, None is returned
+        """
+
+    if (isinstance(term_freq, dict) and isinstance(idf, dict) and term_freq != {}
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:138]: revert the condition and return None immediately

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (NataliaBush) [reply lab_1_keywords_tfidf/main.py:18]: done

File: lab_1_keywords_tfidf/main.py (hunk 3 above)
Comment (NataliaBush) [reply lab_1_keywords_tfidf/main.py:42]: done

File: lab_1_keywords_tfidf/main.py (hunk 5 above)
Comment (NataliaBush) [reply lab_1_keywords_tfidf/main.py:63]: done

File: lab_1_keywords_tfidf/main.py (hunk 2 above)
Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/main.py:11]: No - github shoes that it was changed

File: lab_1_keywords_tfidf/main.py (hunk 6 above)
Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/main.py:123]: .

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**